usertools/sqlite_dictionary.py

```diff
@@ -0,0 +1,162 @@
+import json
+import sqlite3
+import time
+import unicodedata
+
+
+def strip_accents(s: str):
+    return "".join(
+        c for c in unicodedata.normalize("NFD", s) if unicodedata.category(c) != "Mn"
+    )
+
+
+def collate_unicode_no_diacritic_no_case(str1: str, str2: str):
+    # By default sqlite has no unicode case insensitive collation. This additionally removes all accents.
+    # Standard SQLite is not even to search for cyrillic strings without case sensitivity.
+
+    s1 = strip_accents(str1).lower()
+    s2 = strip_accents(str2).lower()
+    if s1 < s2:
+        return -1
+    elif s1 > s2:
+        return 1
+    else:
+        return 0
+
+
+class WiktionaryDictionary:
+    """This class is a wrapper around a sqlite database containing the Wiktionary dump.
```

Comment:
Because of the use of the word "dump", I was confused that this whole thing was about creating a database out of the Wiktionary dump file... That is, the big zip files we get from Wiktionary that contains the source text files. The extracted data should probably be called something else to avoid confusion. `Wiktextract data` or something similar should work.